### PR TITLE
Starred tuple assignment

### PR DESCRIPTION
**Pull request by bcaller**
We can do slightly better than just assigning all of the RHS variables to all the LHS targets.

`a, b = c, d` should be easy to propagate taint to the correct variable for instance.

When we have starargs then we can still try to propagate taint to the variables which could potentially end up tainted.

### CODE REVIEW

File: pyt/cfg/stmt_visitor.py [hunk 1]
```diff
@@ -327,28 +327,59 @@ def visit_Try(self, node):
...
             label = LabelVisitor()
             label.visit(target)
-
+            rhs_visitor = RHSVisitor()
+            rhs_visitor.visit(value)
             if isinstance(value, ast.Call):
                 new_ast_node = ast.Assign(target, value)
-                new_ast_node.lineno = node.lineno
-
+                ast.copy_location(new_ast_node, node)
```
Comment (KevinHock) [inline pyt/cfg/stmt_visitor.py]: Nice, that's awesome.

File: pyt/cfg/stmt_visitor.py [hunk 2]
```diff
@@ -327,28 +327,59 @@ def visit_Try(self, node):
...
+                    rhs_visitor.result,
+                    line_number=node.lineno,
+                    path=self.filenames[-1]
+                )))
+            remaining_targets.remove(target)
+            remaining_values.remove(value)
+            for var in rhs_visitor.result:
+                remaining_variables.remove(var)
+
+        # Pair targets and values until a Starred node is reached
```
Comment (KevinHock) [inline pyt/cfg/stmt_visitor.py]: Woah nice 😮

File: tests/cfg/cfg_test.py [hunk 3]
```diff
@@ -779,6 +781,46 @@ def test_assignment_tuple_value(self):
...
+        self.assertEqual(middle_nodes[0].label, 'a = f')
+        self.assertCountEqual(  # We don't assert a specific order for the assignment nodes
+            [n.label for n in middle_nodes],
+            ['a = f', 'd = f + i', 'e = j'] + ['*b, c = *g, *h'] * 2,
+        )
+        self.assertCountEqual(
+            [(n.left_hand_side, n.right_hand_side_variables) for n in middle_nodes],
+            [('a', ['f']), ('b', ['g', 'h']), ('c', ['g', 'h']), ('d', ['f', 'i']), ('e', ['j'])],
+        )
+
```
Comment (KevinHock) [inline tests/cfg/cfg_test.py]: Possible pep8 nit: w/r/t one line between methods of a class.

**Comment by bcaller** [comment]
Fixed. I'll try and fix travis and tox configs later because I think flake8 should've caused the build to fail.

**Comment by KevinHock** [comment]
Awesome, thanks again :)

**Pull request merged**